Tests/FluentAssertions.Shared.Specs/CollectionEquivalencySpecs.cs

```diff
@@ -17,6 +17,128 @@ namespace FluentAssertions.Specs
     [TestClass]
     public class CollectionEquivalencySpecs
     {
+        #region General
+
+        [TestMethod]
+        public void When_ShouldAllBeEquivalentTo_subject_is_null_it_should_throw()
```

Comment:
> 11. Don't refer to technical names such as the names of methods, variables and UI elements in the name of a test. In my opinion, the name of a test should explain the functional scenario, whereas the cause and effect should clearly illustrate the technicalities of that scenario. If some class or member is renamed, it shouldn't affect the functional name of the test

I suppose the following tests in `CollectionEquivalencySpecs` mislead me:

`When_ShouldAllBeEquivalentTo_has_selection_rules_configured_they_should_be_evaluated_from_right_to_left`
`When_ShouldAllBeEquivalentTo_utilizes_custom_assertion_rules_the_rules_should_be_respected`

I can rename them in addition to the ones I added if you like